### ISSUE TITLE
`lxd-agent` panics in a VM if `security.devlxd=false`

### ISSUE THREAD

**Issue by simondeziel** [canonical/lxd#12629]
Steps to reproduce:

```
$ lxc launch ubuntu-daily:22.04 vm1 --vm -c security.devlxd=false
sleep 30
$ lxc exec --force-noninteractive vm1 -- journalctl --no-hostname -u lxd-agent
Dec 06 16:54:02 systemd[1]: Starting LXD - agent...
Dec 06 16:54:02 systemd[1]: Started LXD - agent.
Dec 06 16:54:03 lxd-agent[234]: 2023/12/06 16:54:03 http: panic serving host(2):9268529: runtime error: invalid memory address or nil pointer dereference
Dec 06 16:54:03 lxd-agent[234]: goroutine 16 [running]:
Dec 06 16:54:03 lxd-agent[234]: net/http.(*conn).serve.func1()
Dec 06 16:54:03 lxd-agent[234]:         net/http/server.go:1868 +0xb9
Dec 06 16:54:03 lxd-agent[234]: panic({0xb25340?, 0x11d1140?})
Dec 06 16:54:03 lxd-agent[234]:         runtime/panic.go:920 +0x270
Dec 06 16:54:03 lxd-agent[234]: net/http.(*Server).Close(0x0)
Dec 06 16:54:03 lxd-agent[234]:         net/http/server.go:2737 +0x37
Dec 06 16:54:03 lxd-agent[234]: main.stopDevlxdServer(0xd2ea60?)
Dec 06 16:54:03 lxd-agent[234]:         github.com/canonical/lxd/lxd-agent/api_1.0.go:172 +0x93
Dec 06 16:54:03 lxd-agent[234]: main.api10Put(0xc000099a00, 0x0?)
Dec 06 16:54:03 lxd-agent[234]:         github.com/canonical/lxd/lxd-agent/api_1.0.go:120 +0x28c
Dec 06 16:54:03 lxd-agent[234]: main.createCmd.func1.1({0xc0e390?})
Dec 06 16:54:03 lxd-agent[234]:         github.com/canonical/lxd/lxd-agent/server.go:76 +0x26
Dec 06 16:54:03 lxd-agent[234]: main.createCmd.func1({0xd2d730, 0xc000118b60}, 0xc00011af00)
Dec 06 16:54:03 lxd-agent[234]:         github.com/canonical/lxd/lxd-agent/server.go:83 +0x379
Dec 06 16:54:03 lxd-agent[234]: net/http.HandlerFunc.ServeHTTP(0xc00011ae00?, {0xd2d730?, 0xc000118b60?}, 0x1?)
Dec 06 16:54:03 lxd-agent[234]:         net/http/server.go:2136 +0x29
Dec 06 16:54:03 lxd-agent[234]: github.com/gorilla/mux.(*Router).ServeHTTP(0xc000000240, {0xd2d730, 0xc000118b60}, 0xc00011ad00)
Dec 06 16:54:03 lxd-agent[234]:         github.com/gorilla/mux@v1.8.1/mux.go:212 +0x1c5
Dec 06 16:54:03 lxd-agent[234]: net/http.serverHandler.ServeHTTP({0xd2b378?}, {0xd2d730?, 0xc000118b60?}, 0x6?)
Dec 06 16:54:03 lxd-agent[234]:         net/http/server.go:2938 +0x8e
Dec 06 16:54:03 lxd-agent[234]: net/http.(*conn).serve(0xc000120ea0, {0xd2ec20, 0xc0001b8a80})
Dec 06 16:54:03 lxd-agent[234]:         net/http/server.go:2009 +0x5f4
Dec 06 16:54:03 lxd-agent[234]: created by net/http.(*Server).Serve in goroutine 6
Dec 06 16:54:03 lxd-agent[234]:         net/http/server.go:3086 +0x5cb
```

Version info:

```
$ snap list lxd
Name  Version      Rev    Tracking     Publisher   Notes
lxd   git-fdc3dc6  26461  latest/edge  canonical✓  -

$ lxc config show vm1
architecture: x86_64
config:
  image.architecture: amd64
  image.description: ubuntu 22.04 LTS amd64 (daily) (20231201)
  image.label: daily
  image.os: ubuntu
  image.release: jammy
  image.serial: "20231201"
  image.type: disk-kvm.img
  image.version: "22.04"
  security.devlxd: "false"
  volatile.base_image: c3ddddd67a47a471aca0d330cd09b899c12e82e48eddf7ec79777b563ea26bb1
  volatile.cloud-init.instance-id: fea5bac3-4fe0-4498-9070-4f1a3a58f700
  volatile.eth0.host_name: tap11b752af
  volatile.eth0.hwaddr: 00:16:3e:f5:58:26
  volatile.last_state.power: RUNNING
  volatile.uuid: f533ebd0-3383-48c5-8556-df3dc6caf13f
  volatile.uuid.generation: f533ebd0-3383-48c5-8556-df3dc6caf13f
  volatile.vsock_id: "381798246"
devices: {}
ephemeral: false
profiles:
- default
stateful: false
description: ""
```

**Comment by MusicDin** [comment]
I've managed to reproduced it. Thanks.

**Comment by tomponline** [comment]
@MusicDin lets get this fix into LXD 5.20 too please
